streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -642,21 +642,7 @@ abstract class ZStream[-R, +E, +O](
         left  <- self.process.mapM(BufferedPull.make[R, E, O](_)) // type annotation required for Dotty
         right <- that.process.mapM(BufferedPull.make[R1, E1, O2](_))
         pull <- ZStream
-                 .unfoldM(s) { s =>
-                   f(s, left.pullElement, right.pullElement).flatMap { exit =>
-                     ZIO.done(
-                       exit.fold(
-                         Cause
-                           .sequenceCauseOption(_) match {
-                           case None    => Exit.succeed(None)
-                           case Some(e) => Exit.halt(e)
-                         },
-                         result => Exit.succeed(Some(result))
-                       )
-                     )
-
-                   }
-                 }
+                 .unfoldM(s)(s => f(s, left.pullElement, right.pullElement).flatMap(ZIO.done(_).optional))
```

Comment:
Nice!!!! We can probably apply this in a bunch of other places throughout the package :-)